plugin/service/storage/plugin.go

```diff
@@ -820,7 +820,7 @@ func dryRunValidation(ctx context.Context, state *awsStoragePersistedState, attr
 		st, permissions.Read = errors.ParseAWSError(err, "failed to list object")
 		errs = multierror.Append(errs, st.Err())
 	} else if res == nil || len(res.Contents) != 1 || *res.Contents[0].Key != objectKey {
-		return status.New(codes.Aborted, fmt.Sprintf("list response did not contain the expected key: %+v", res))
+		errs = multierror.Append(errs, status.New(codes.Aborted, fmt.Sprintf("list response did not contain the expected key: %+v", res)).Err())
```

Comment:
Should the `permissions.Read` be set to `Unknown`?

Reply:
This is equivalent to [NoSuchKey](https://github.com/aws/aws-sdk-go-v2/blob/4599f78694cabb6853addabc6f92cb197fdb5647/service/s3/types/errors.go#L123) where we want clients to retry

Reply:
The permission is set to unknown if the key does not exist